pkg/security/probe/perf_buffer_monitor.go

```diff
@@ -376,60 +411,62 @@ func (e *EventsStats) collectAndSendKernelStats(client *statsd.Client) error {
 				}
 
 				// Update stats to avoid sending twice the same data points
-				if tmpCount = e.swapKernelEventBytes(evtType, perfMapName, cpu, stats.Bytes); tmpCount <= stats.Bytes {
+				if tmpCount = pbm.swapKernelEventBytes(evtType, perfMapName, cpu, stats.Bytes); tmpCount <= stats.Bytes {
 					stats.Bytes -= tmpCount
 				}
-				if tmpCount = e.swapKernelEventCount(evtType, perfMapName, cpu, stats.Count); tmpCount <= stats.Count {
+				if tmpCount = pbm.swapKernelEventCount(evtType, perfMapName, cpu, stats.Count); tmpCount <= stats.Count {
 					stats.Count -= tmpCount
 				}
-				if tmpCount = e.swapKernelLostCount(evtType, perfMapName, cpu, stats.Lost); tmpCount <= stats.Lost {
+				if tmpCount = pbm.swapKernelLostCount(evtType, perfMapName, cpu, stats.Lost); tmpCount <= stats.Lost {
 					stats.Lost -= tmpCount
 				}
 
 				if client != nil {
-					if err := e.sendKernelStats(client, stats, tags); err != nil {
+					if err := pbm.sendKernelStats(client, stats, tags); err != nil {
 						return err
 					}
 				}
+				total += stats.Lost
+				perEventPerCPU[evtType.String()][cpu] += stats.Lost
 			}
-
 		}
 		if iterator.Err() != nil {
 			return errors.Wrapf(iterator.Err(), "failed to dump the statistics buffer of map %s", perfMapName)
 		}
+
+		// send an alert if events were lost
+		if total > 0 {
+			pbm.probe.DispatchCustomEvent(
+				NewEventLostWriteEvent(perfMapName, perEventPerCPU, time.Now()),
```

Comment:
same here I will `time.Now()` in the marshall function

Reply:
👍🏻